### PR TITLE
added directive needattribute

### PR DESCRIPTION
**Pull request by PhilipPartsch**
try to implement #613
Have to do a lot, but try to get a first feedback.

### CODE REVIEW

File: sphinx_needs/layout.py [hunk 1]
```diff
@@ -456,7 +455,28 @@ def meta(self, name: str, prefix: Optional[str] = None, show_empty: bool = False
...
+            ismultiline = self.need["attribute_metadata"][name]["multiline"]
+        except:
+            ismultiline = False
+
+        try:
+            isuml = self.need["attribute_metadata"][name]["uml"]
+        except:
+            isuml = False
+
+        if isuml:
```
Comment (PhilipPartsch) [inline sphinx_needs/layout.py:468]: uml part has to be shifted from needuml. It would be nice to refer to uml with <need id>.attibute.

**Review by danieleades — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -86,6 +86,7 @@ module = [
   'sphinx_needs.builder',
   'sphinx_needs.diagrams_common',
   'sphinx_needs.directives.need',
+  'sphinx_needs.directives.needattribute',
```
[the remark this answers is not in the document]
Comment (PhilipPartsch) [reply pyproject.toml]: Thnaks for the hint, changed it.

**Comment by danwos** [comment]
I like the idea of ``needattribute``, but this PR also contains changes to ``needuml``, which should not be part of this PR and makes things complicated.

We should concentrate on storing the data first. Then, in further PRs, we deal with using this data in some other directives.

For me also some parts are looking more use case specific and are not generic enough.
But I will make a quick review and add my notes :)

For next PRs it would be also a good idea to write down a small concept with some rest example code first. 
So we can discuss it and get on the same page.

**Review by danwos — CHANGES_REQUESTED** [review]
Puhh, in this PR is happening so much and a lot of new features are added.
But all with no concept description or use case discussion.

I really would like to see the following changes in separate issues:
- needattribute
- Refactoring needuml (what about needflow?)
- multiline support in meta-data / layouts
- uml support in meta data / layouts

Especially for uml-support I have some doubts and see different solutions to get specific use-cases done.

Don't get me wrong, I like most of the changes, but I'm somehow lost seeing the impact to different use cases.

BTW: Isn't ``needattribute`` somehow the little sister of ``needextend``.
Maybe we can combine them....

File: sphinx_needs/directives/needattribute.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        targetid = f"needattribute-{env.docname}-{id}"
+        error_id = f"Needattribute - file '{env.docname}' - line '{self.lineno}'"
+
+        content = self.content
+        if not content:
+            raise Exception(f"{error_id} content cannot be empty.")
+
+        #content = "\n".join(content) 
+
+        title = self.arguments[0].strip() if self.arguments else None
```
[the remark this answers is not in the document]
Comment (PhilipPartsch) [reply sphinx_needs/directives/needattribute.py:69]: I did not fought about his. I use the None normally to know about unset values and excpetion creation if need.

File: sphinx_needs/layout.py (hunk 1 above)
Comment (PhilipPartsch) [reply sphinx_needs/layout.py:468]: I implemented a protoype to provide an exmaple requested [here](https://github.com/useblocks/sphinxcontrib-needs/issues/613#issuecomment-1160583656)

**Pull request closed without merging**